geoAssembler/optimiser/centre.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+
+from collections import namedtuple
+from itertools import product
+
+from .utility import Integrator
+
+from extra_geom.detectors import DetectorGeometryBase
+from typing import Union
+
+
+class CentreOptimiser:
+    """
+    Object used to manage the optimisation of the centre position
+    via maximization of the azimuthal integration result.
+
+    The centre position leading to the highest azimuthal integration
+    value at a radial bin should be the one where the diffraction
+    rings form as straight a line as possible, and the rings becoming
+    straight lines in polar coordinates indicates an accurate centre.
+    """
+    def __init__(self, geom: DetectorGeometryBase,
+                 frame: np.ndarray, sample_dist_mm: Union[int, float],
+                 unit: str = "2th_deg"):
+        self.frame = frame
+        self.integrator = Integrator(geom, sample_dist_mm, unit)
+        self.integrate2d = self.integrator.integrate2d
+
+        #  Slightly dodgy way to pull the quadrant corner positions out of geom
+        #  TODO: Suggest adding this in to extra-geom?
```

Comment:
Yes, I must remember to figure out how to add a notion of quadrant positions in EXtra-geom.

In the meantime, can you use `/ geom.pixel_size` instead of `* geom._get_plot_scale_factor('px')`? That looks a little bit less hacky.